themes/default/content/blog/next-level-iac-pulumi-runtime-logic/index.md

```diff
@@ -0,0 +1,428 @@
+---
+title: "Next-level IaC: Do more with languages!"
+date: 2024-04-26
+meta_desc: Learn how to use your language of choice to do more than simply declare cloud resources.
+meta_image: meta.png
+authors:
+    - christian-nunciato
+tags:
+    - next-level-iac
+    - nodejs
+    - python
+    - cloudformation
+    - terraform
+    - hcl
+---
+
+Our users are always telling us (particularly the ones who come to Pulumi from other IaC tools) that being able to use general-purpose languages to manage their infrastructure was a game changer for them.
+
+I know it was for me. As a JavaScript developer, when I discovered Pulumi and saw that I could do pretty much everything I was doing with Terraform _but with TypeScript_, I was immediately hooked; that's all it took. For me, just being able to write my resource declarations in a full-featured language that I knew well, and that my IDE deeply understood, was huge.
+
+But it's easy to forget, even when you've been using Pulumi for a while, that you aren't just getting the language alone --- you're getting the full breadth and capability of that language's runtime environment as well. Not just JavaScript, but all of Node.js. Not just C#, but everything the .NET framework and common-language runtime have to offer.
+
+This is quite a big deal, actually. Whereas with other tools, you're bound by the limitations of the tool's DSL (CloudFormation YAML, Terraform HCL, etc.), with Pulumi, you have no such constraints; you can do just about anything your chosen language and runtime can do. That means much more than just declare long lists of cloud resources --- it means unlocking all sorts of capabilities that allow you do more with less, handle change easily, and really bring your infrastructure-management practice to a whole new level.
+
+In this post, the first of a series, we'll show you a few things you can do to take advantage of this reality.
```

Comment:
i would actually list out the X number of things you plan to go through rather than just saying a few